spec/services/csv/sales_log_csv_service_spec.rb

```diff
@@ -3,23 +3,27 @@
 RSpec.describe Csv::SalesLogCsvService do
   let(:form_handler_mock) { instance_double(FormHandler) }
   let(:organisation) { create(:organisation) }
-  let!(:log) { create(:sales_log, :completed, owning_organisation: organisation, purchid: nil) }
+  let(:fixed_time) { Time.zone.local(2023, 2, 8) }
+  let(:user) { create(:user, email: "billyboy@eyeKLAUD.com") }
+  let!(:log) do
+    create(
+      :sales_log,
+      :completed,
+      created_by: user,
+      saledate: fixed_time,
+      created_at: fixed_time,
+      updated_at: fixed_time,
+      owning_organisation: organisation,
+      purchid: nil,
+    )
+  end
   let(:service) { described_class.new(export_type: "labels") }
   let(:csv) { CSV.parse(service.prepare_csv(SalesLog.all)) }
 
-  around do |example|
-    Timecop.freeze(Time.utc(2023, 2, 8)) do
-      Singleton.__init__(FormHandler)
-      example.run
-    end
-    Timecop.return
-    Singleton.__init__(FormHandler)
-  end
-
   it "calls the form handler to get all questions in order when initialized" do
     allow(FormHandler).to receive(:instance).and_return(form_handler_mock)
     allow(form_handler_mock).to receive(:ordered_sales_questions_for_all_years).and_return([])
-    described_class.new(export_type: "codes")
+    service
```

Comment:
How come we need this?

Reply:
just that this initialises the service, and as part of that we call the form handler